\documentclass{article}

\usepackage[T1]{fontenc}
\usepackage{hyperref}
\usepackage{longtable}
\usepackage{booktabs}
\usepackage{pdfpages}
\usepackage{lineno}
\usepackage{todonotes}

\hypersetup{
    colorlinks=true,       % false: boxed links; true: colored links
    urlcolor=blue           % color of external links
}

% Comments begin with a percent.
% define the title
\author{Brian J. Knaus, Javier F. Tabima and Niklaus J. Gr\"{u}nwald}
\title{Mitochondrial molecular markers for US lineages of \emph{P. infestans}}

% Begin the document.
\begin{document}
%\SweaveOpts{concordance=TRUE}

% generates the title
\maketitle
\newpage

% Abstract
%\begin{abstract}
%Morphometric analysis of \emph{Carex minicaulis}.
%\vspace{8pt}
%Keywords: \emph{Carex}; Morphometrics; Taxonomy.
%\end{abstract}
%\pagebreak

% insert the table of contents
\tableofcontents
\listoffigures
\listoftables

\newpage

%\newpage

%% ---------- ---------- ---------- Section ---------- ---------- ---------- %%

\linenumbers

\section{The sample}

The sample includes data which was opportunistically gathered from previous publications as well as data which is not yet available to the public (Judelson, unpublished).

\vspace{12pt}

\begin{center}
\begin{tabular}{ l c c }
  \hline
  Sample & Count & Reference \\
  \hline
  T30-4 & 1 & \cite{haas2009genome} \\
  PIC99189 \& 90128 & 2 & \cite{raffaele2010analyses} \\
  13\_a2 & 1 & \cite{cooke2012genome} \\
  Yoshida et al. & 13 & \cite{yoshida2013correction} \\
  Martin et al. & 3 & \cite{martin2013reconstructing} \\
  Judelson & 8 & NA \\
  \hline
  Total & 28 & \\
  \hline
\end{tabular}
\end{center}

\vspace{12pt}

\begin{itemize}

\item The sample `T30-4' was the first sequenced genome and is considered the reference for nuclear work \cite{haas2009genome}.  This genome was assembled prior to high-throughput sequencing (i.e., Illumina and 454 technologies).  The data presented here are not the sequences used for the paper but are part of a project by \href{http://www.broadinstitute.org/annotation/genome/phytophthora_infestans/MultiHome.html}{The Broad} to resequence this individual using Illumina and 454 technologies.

\item Note that both the Yoshida and Martin papers included ancient DNA in their analyses \cite{yoshida2013correction, martin2013reconstructing}.  Here we have omitted those samples and focused on modern samples.

\item For enigmatic reasons, not all of the samples from the Yoshida and Martin papers were actually available online.  Therefore our numbers here do not match those presented in the papers.  

\item The Judelson data include a sample of US1 which was sampled at three different time points (us1\_1, us1\_2 and us1\_3).  We suspect that these were different samples and not necessarily the same clone.  Therefore differences among these samples may either be due to biological or technical factors.

\item The Judelson data includes a sample of US8 which has been characterized as having fungicide resistance \cite{danies2013phenotypic}.  This lineage was also sequenced by Martin et al. \cite{martin2013reconstructing}.  These are most likely different samples so differences among these samples may be interpreted as biological.  
  
\item The Yoshida data includes one sample of \emph{P. mirabilis} (p7722), this should jump out in the analyses.

\item For the mitochondrial data we used the type IIa form \cite{avila2006mitochondrial} because it was the longest sequence and we felt this would provide the best alignment.

\end{itemize}

We've used the term `SNP' fairly loosely in this document.  The term `variant' may be more appropriate.  Until fairly recently the software tools we've been using could only handle SNPs.  They now report short indels as well.  We've included both variant types here.

%% ---------- ---------- ---------- Section ---------- ---------- ---------- %%

\section{Variant discovery}

Reads were mapped to the type IIa mitochondrial reference \href{http://www.ncbi.nlm.nih.gov/nuccore/AY898627.1}{"AY898627.1"}.  Reads were mapped using bowtie2 \cite{langmead2012fast}.  Variants were called using SAMtools\cite{li2009sequence}.

\subsection{Variant filtering}
As a quality control step, the variant files were filtered by quality, read depth and mapping quality (Figures \ref{fig:plot_unfilter},\ref{fig:plot_filter}). For this we used an in-house R package called \href{https://github.com/knausb/vcfR}{vcfR}. Here, sequencing depth is cumulative over all samples.  Quality here is for each variant over all samples and ranges from 1-999.

The genotype caller in Samtools assumes a diploid, bi-allelic model.  Because mitochondria are assumed to be haploid we tried to filter out heterozygous calls.  Samples which included high quality heterozygote calls (p1362, p6096, p10650, p12204, p10127) were mostly from the Yoshida et al. \cite{yoshida2013correction} paper and were among the low sequencing depth samples they included.  Because these samples are not among the US lineages we're interested in, and because they are apparently of low sequencing depth, we omitted them for now.  However, the sample nl07434 was among the high sequencing depth samples from this paper and is perhaps noteworthy.  T30-4 was called as a heterozygote for one variant and is perplexinng.

In an attept to identify high quality variants we employed a filtering strategy.  Filtering of the variant panel was based on quality (QUAL=999), cumulative sequencing depth (1st quartile >= DP >= 3rd quartile) and mapping quality (1st quartile >= MQ >= 3rd quartile).  This resulted in 37 variants remaining after filtering (Table \ref{ptab}).  We have identified a fraction of these as being diagnostic for a small group of samples (Table \ref{table:DiagSNP}).

The variants remaining after filtering were visualized as a linear chromosome in Figure \ref{fig:chromogc}.


<<vcfContruct, echo=FALSE, results='hide'>>=
library(vcfR)
data(vcfR_example)
mt <- read.vcf('pinf_1.raw.vcf.gz')
pinf_mt <- create.chrom('pinf_mt', seq=pinf_dna, vcf=mt, ann=pinf_gff)
@

<<echo=FALSE, plot_unfilter, comment='', fig.lp="fig:", fig.cap = "Quality control results for the mtDNA SNP calls before filtering.", fig.pos='p'>>=
plot(pinf_mt)
@

<<mask, echo=FALSE, results='hide'>>=
pinf_mt <- masker(pinf_mt)
pinf_mt <- proc.chrom(pinf_mt)
@

<<echo=FALSE, plot_filter, comment='', fig.lp="fig:", fig.cap = "Quality control results for the mtDNA SNP calls after filtering and windowizing variants.", fig.pos='p'>>=
plot(pinf_mt)
@

<<chromR, echo=FALSE>>=
#
# These are mitochondria, so they should be haploid.
# Let's make sure.
#gt <- t(apply(pinf_mt@vcf.gt, MARGIN=1, function(x){unlist(lapply(strsplit(as.character(x), ":"), function(x){x[1]}))}))

get.gt1 <- function(x, element="GT"){
  FORMAT <- unlist(strsplit(as.character(x[1]), ":"))
  x <- x[-1]
  pos <- grep(element, FORMAT)
  unlist(lapply(strsplit(as.character(x), ":"), function(x){x[pos]}))
}
get.gt <- function(x, element="GT"){
  gt <- t(apply(x@vcf.gt, MARGIN=1,get.gt1, element=element))
  colnames(gt) <- names(x@vcf.gt)[-1]
  gt
}
gt <- get.gt(pinf_mt, element="GT")
gq <- get.gt(pinf_mt, element="GQ")
#
# Validate we got the genotypes correctly.
#gt <- gt[,-1]
gt[gt=="0/0"] <- 0
gt[gt=="0/1"] <- 1
gt[gt=="1/1"] <- 2
#
gt.m <- matrix(ncol=ncol(gt), nrow=nrow(gt))
gq.m <- matrix(ncol=ncol(gt), nrow=nrow(gt))
for(i in 1:ncol(gt)){
  gt.m[,i] <- as.numeric(gt[,i])
  gq.m[,i] <- as.numeric(gq[,i])
}
colnames(gt.m) <- colnames(gt)
colnames(gq.m) <- colnames(gq)
pinf_mt@var.info$mask[rowSums(apply(gt.m, MARGIN=2, function(x){x==1})) != 0] <- FALSE
#sum(pinf_mt@mask)
#
pinf_mt <- proc.chrom(pinf_mt,verbose=FALSE)
#
write.vcf(pinf_mt, "pinf_mt_masked.vcf")
cat("Before filtering:")
length(pinf_mt@var.info$mask)
# After filter
cat("After filtering:")
sum(pinf_mt@var.info$mask)
@

<<echo=FALSE, variants, results='asis'>>=
#library(knitr)
#kable(pinf_mt@vcf.fix[pinf_mt@mask,c(1,2,4,5)])
#kable(pinf_mt@vcf.fix[1:4,c(1,2,4,5)])
#kable(head(iris))
#
library(xtable)
ptab <- xtable(cbind(pinf_mt@vcf.fix[pinf_mt@var.info$mask,c(1,2,4,5)]))
#colnames(ptab)[5] <- "Tree Number"
align(ptab) <- 'llccc'
caption(ptab) <- "Variants remaining after filtering."
label(ptab) <- "ptab"
digits(ptab)[3] <- 0
ptab
@


<<echo=FALSE, comment='', chromogc, fig.lp="fig:", fig.cap = "Whole mtDNA genome scan for the P. infestans samples.",  fig.pos='p', results='hide'>>=
chromoqc(pinf_mt)
@


%\section{SNPs after filtering}



%After filtering we proceeded to look at which of the SNP's were supported after the filtering (Table \ref{table:DiagSNP}).

\begin{table}[h!]
\centering
\caption{Diagnostic SNP positions for the mtDNA genome after filtering. The tree number corresponds to the character legend on the tree figures}
\begin{tabular}{@{}llll@{}}
\toprule
Position & SNP & Diagnostic for & Tree Number      \\
\midrule
7857     & A/G & p10127, us11 and us1 & 4\\
8118     & T/G & us1                  & 5\\
20464    & T/C & p17777, us22 and us8 & 15\\
22711    & G/A & p17777, us22 and us8 & 17\\
26767    & T/C & p10127, us11 and us1 & 24\\
28783    & A/G & p10127, us11 and us1 & 26\\
36663    & T/C & us8 & 33                  \\
\bottomrule
\end{tabular}
\label{table:DiagSNP}
\end{table}

%% ---------- ---------- ---------- Section ---------- ---------- ---------- %%

\section{Variant segregation}
%\section{Diagnostic SNP mapping}

In order to visualize how variants segregated among the samples, a phylogeny was inferred.  We then used ancestral state reconstruction to map the characters to the tree.  At this time we're not trying to say anything bold about phylogeny or character evolution.  We're simply using these tools to visualize how the variants segregate.

\subsection{Phylogenetic reconstruction}
Using the whole genome alignment (28 sequences, 39,870 nucleotides) we performed a whole-genome phylogeny using maximum likelihood (RAxML) and Bayesian inference (BEAST). We used RAxML using no partitions, 1000 bootstrap replicates, a \texttt{GTR+I+G} model of nucleotide evolution to obtain a bipartitioned tree with the boostrap values mapped to the branches. For BEAST, we specified \texttt{p7722} (\emph{P.mirabilis}) as the outgroup. We used a \texttt{HKY+G+I} model of nucleotide substitutions, a strict molecular clock, a constant population size prior, UPGMA starting tree and 10 million Markov chains. The best tree is shown in Figure \ref{fig:BEAST}.

\begin{figure}[p]
\centering
%\includegraphics[scale=0.5]{mtdna_alin.jpg}
\includegraphics[width=1.0\textwidth]{mtdna_alin.jpg}
\caption[BEAST Coalescent tree]{Bayesian coalescent tree of the whole mtDNA genome of \emph{P. infestans} using \texttt{BEAST 1.8.0}. Values above branches represent branch lengths (topology is concordant to the bifurcating model of a coalescent reconstruction). Branches are colored based on their posterior probability values (legend indicates color scheme).}
\label{fig:BEAST}
\end{figure}

\subsection{Mapping the SNP's in the BEAST tree}

To map the variants found in the mtDNA genome to the coalescent tree, we used \texttt{Mesquite}. We did a removal of invariable regions and ancestral state reconstruction for all 37 SNPs using a parsimony reconstruction state (see tree figures).

\newpage



\includepdf[pages={5,6,16,18,25,27,34}, nup=2x2]{All_traces.pdf}


\vspace{24pt}

%% ---------- ---------- ---------- Section ---------- ---------- ---------- %%

\section{Session information}

<<sesinf>>=
sessionInfo()
@

%% ---------- ---------- ---------- Section ---------- ---------- ---------- %%

\bibliographystyle{plain}
\bibliography{lineages.bib}

%% ---------- ---------- ---------- Section ---------- ---------- ---------- %%

\end{document}